src/app/pages/test-report/cat-adi-part3/components/adi3-end-test-modal/adi3-end-test-modal.ts

```diff
@@ -16,18 +17,22 @@ export class Adi3EndTestModal implements OnInit {
   totalScore: number;
   feedback: string;
   isValidDashboard: boolean;
+  isTestReportPopulated: boolean;
 
   constructor(
     private modalCtrl: ModalController,
     private navParams: NavParams,
-  ) {}
+    private adi3AssessmentProvider: ADI3AssessmentProvider,
+  ) {
+  }
 
   ngOnInit(): void {
     this.testData = this.navParams.get('testData');
     this.testResult = this.navParams.get('testResult');
     this.totalScore = this.navParams.get('totalScore');
     this.feedback = this.navParams.get('feedback');
     this.isValidDashboard = this.navParams.get('isValidDashboard');
+    this.isTestReportPopulated = this.adi3AssessmentProvider.isTestReportPopulated(this.testData);
```

Comment:
I think we use navParams/inputs like shown above so we keep all of the business logic inside the higher level components and then just feed in a value.

You should be able to create a isTestReportPopulated in the ```onContinueClick``` method found ```src/app/pages/test-report-dashboard/test-report-dashboard.page.ts```.